src/replay/service.zig

```diff
@@ -30,26 +32,65 @@ const LatestValidatorVotes = sig.consensus.latest_validator_votes.LatestValidato
 const SlotHistoryAccessor = sig.consensus.replay_tower.SlotHistoryAccessor;
 const VoteListener = sig.consensus.vote_listener.VoteListener;
 
+const ProcessResultState = replay.process_result.ProcessResultState;
 const ReplayExecutionState = replay.execution.ReplayExecutionState;
-const SlotTracker = replay.trackers.SlotTracker;
+const ReplayResult = replay.execution.ReplayResult;
+
 const EpochTracker = replay.trackers.EpochTracker;
+const SlotTracker = replay.trackers.SlotTracker;
+const SlotTree = replay.trackers.SlotTree;
 
+const awaitResults = replay.execution.awaitResults;
+const processResult = replay.process_result.processResult;
 const updateSysvarsForNewSlot = replay.update_sysvar.updateSysvarsForNewSlot;
 
 const LatestValidatorVotesForFrozenSlots =
     sig.consensus.latest_validator_votes.LatestValidatorVotes;
 
 pub const Logger = sig.trace.Logger("replay");
 
-/// Number of threads to use in replay's thread pool
-const NUM_THREADS = 4;
-
 const SWITCH_FORK_THRESHOLD: f64 = 0.38;
 const MAX_ENTRIES: u64 = 1024 * 1024; // 1 million slots is about 5 days
 const DUPLICATE_LIVENESS_THRESHOLD: f64 = 0.1;
 pub const DUPLICATE_THRESHOLD: f64 = 1.0 - SWITCH_FORK_THRESHOLD - DUPLICATE_LIVENESS_THRESHOLD;
 
-pub const ReplayDependencies = struct {
+pub const Service = struct {
+    replay: ReplayState,
+    consensus: ?ConsensusState,
+    num_threads: u32,
+
+    pub fn init(
+        deps: Dependencies,
+        enable_consensus: ?ConsensusState.Dependencies,
+        num_threads: u32,
+    ) !Service {
+        var state = try ReplayState.init(deps, num_threads);
```

Comment:
nit:

```suggestion
        var replay_state = try ReplayState.init(deps, num_threads);
        errdefer state.deinit();

        var consensus_state: ?ConsensusState = if (enable_consensus) |consensus_deps|
            try ConsensusState.init(deps, consensus_deps, &state)
```

At the risk of slight verbosity. This makes the naming consistent (as it is not clear why the state for replay is only called state, while state for consensus is called consensus)